pkg/clustermesh/types/types.go

```diff
@@ -38,25 +38,15 @@ type CiliumClusterConfigCapabilities struct {
 	SyncedCanaries bool `json:"syncedCanaries,omitempty"`
 }
 
-func (c0 *CiliumClusterConfig) IsCompatible(c1 *CiliumClusterConfig) error {
-	if c1 == nil {
-		// When remote cluster doesn't have cluster config, we
-		// currently just bypass the validation for compatibility.
-		// Otherwise, we cannot connect with older cluster which
-		// doesn't support cluster config feature.
-		//
-		// When we introduce a new cluster config can't be ignored,
-		// we should properly check it here and return error. Now
-		// we only have ClusterID which used to be ignored.
+func (c *CiliumClusterConfig) Validate() error {
+	if c == nil {
 		return nil
-	} else {
-		// Remote cluster has cluster config. Do validations.
+	}
 
-		// ID shouldn't be duplicated
-		if c0.ID == c1.ID {
-			return fmt.Errorf("duplicated cluster id")
-		}
+	if err := ValidateClusterID(c.ID); err != nil {
+		return err
```

Comment:
This check is potentially problematic in the kvstoremesh case. Indeed, kvstoremesh could connect to a remote cluster which does not expose the cluster config. In that case, it would propagate a cluster config with the appropriate capabilities set, but the cluster ID unset (i.e., = 0). Hence, this check would fail.

Reply:
Remove id = 0 case from validation

Reply:
I'd personally not drop the id == 0 check from `ValidateClusterID`, because it might break the expectations of already existing callers of that function. What about changing the check above:

```diff
--- if c == nil {
+++ if c == nil || c.ID == 0 {
```

At the end of the day, they are equivalent (they both essentially mean that the cluster ID is unknown), and if one is permitted also the other should be.